src/main/kotlin/fi/metatavu/rapurc/api/impl/V1ApiImpl.kt

```diff
@@ -1398,42 +1430,156 @@ class V1ApiImpl : V1Api, AbstractApi() {
+            userId = userId
+        )
+
+        return createOk(groupJoinInviteTranslator.translate(createdInvite))
     }
 
-    @RolesAllowed(value = [ UserRole.USER.name ])
-    override fun findGroupJoinInvite(groupId: UUID?, inviteId: UUID?): Response {
-        TODO("Not yet implemented")
+    @RolesAllowed(value = [ UserRole.USER.name, UserRole.ADMIN.name ])
+    override fun findGroupJoinInvite(groupId: UUID, inviteId: UUID): Response {
+        val userId = loggedUserId ?: return createUnauthorized(NO_LOGGED_USER_ID)
+        val currentUserEmail =  keycloakController.findUserById(userId) ?: return createUnauthorized(NO_LOGGED_USER_ID)
+
+        val groupJoinRequest = groupJoinController.findGroupJoin(
+            groupId = groupId,
+            type = JoinRequestType.INVITE,
+            requestId = inviteId
+        ) ?: return createNotFound(createNotFoundMessage(target = GROUP_JOIN_INVITE, id = inviteId))
+
+        // if it's not the invited user viewing the invite, check if it's group's admin
+        if (groupJoinRequest.email != currentUserEmail.email) {
+            groupAdminAccessRightsCheckFail(userId, groupId)?.let { return it }
+        }
+
+        return createOk(groupJoinRequestTranslator.translate(groupJoinRequest))
     }
 
-    @RolesAllowed(value = [ UserRole.USER.name ])
-    override fun sendGroupJoinInviteEmail(groupId: UUID?, inviteId: UUID?): Response {
-        TODO("Not yet implemented")
+    @RolesAllowed(value = [ UserRole.USER.name, UserRole.ADMIN.name ])
+    override fun sendGroupJoinInviteEmail(groupId: UUID, inviteId: UUID): Response {
+        val userId = loggedUserId ?: return createUnauthorized(NO_LOGGED_USER_ID)
+        groupAdminAccessRightsCheckFail(userId, groupId)?.let { return it }
+
+        val existingInvite = groupJoinController.findGroupJoin(
+            groupId = groupId,
+            type = JoinRequestType.INVITE,
+            requestId = inviteId
+        ) ?: return createNotFound(createNotFoundMessage(target = GROUP_JOIN_INVITE, id = inviteId))
+
+        if (existingInvite.groupId != groupId) {
+            return createNotFound(createNotFoundMessage(target = GROUP_JOIN_INVITE, id = inviteId))
+        }
+
+        val group = keycloakController.findGroupById(groupId) ?: return createNotFound(createNotFoundMessage(target = USER_GROUP, id = groupId))
+        val groupAdmin = keycloakController.findGroupAdmin(group) ?: return createNotFound(createNotFoundMessage(target = GROUP_ADMIN, id = groupId))
+        val invitedUser = keycloakController.findUserById(userId) ?: return createNotFound(createNotFoundMessage(target = USER, id = userId))
+
+        groupJoinController.informUser(
+            user = invitedUser,
+            admin = groupAdmin,
+            subject = "You were invited to join group ${group.name}",
+            body = "You were invited to join group ${group.name}. Please log in to the system to accept or reject the invitation."
```

Comment:
Most likely this will need some sort of html template but that can be done in separate issue and branch